### DISCUSSION TITLE
Finalize the GraphQL API: Support permissions, mutations and subscriptions

### DISCUSSION THREAD

**Discussion by martin-braun** [directus/directus#6362] [Feature Requests]
Quote from the [Directus Docs](https://v8.docs.directus.io/api/graphql.html#introduction):

> Admin Role Only
> 
> Due to the complexity of handling permissions, as of now the GraphQL endpoint can only be used with Admin role accounts. An admin access token needs to be passed in the `access_token` query parameter.

> Limited Support
> 
> Currently supports "Queries" only. "Mutations" and "Subscription" support will be added at a later date.

I'm sure the team is already working on it or at least planning to improve the GraphQL API, but I still want to create this "Idea" in the discussions area of this GitHub repository to keep track of the process. 

I prefer directus over other headless CMS a-likes for its open-source nature, so I think it would be great if directus evolves further in this direction. Being able to use the GraphQL API with permissions and mutations is mandatory for me to consider directus as the "Holy Grail" for headless data management.

I'd love to see the devs sharing their progress on the GraphQL API, here.

**Comment by benhaynes** [comment]
That link is from the Directus 8 docs, please see the new Directus 9 docs here: https://docs.directus.io/reference/api/introduction/

**Reply by martin-braun** [reply]
I'm literally facepalming myself. I was given this link by duckduckgo when searching for "directus graphql". Should've used Google right away. Maybe it would make sense to add a banner on top of the v8 docs pointing to the recent docs?

**Reply by benhaynes** [reply]
Added : https://github.com/directus/v8-archive/blob/master/docs/getting-started/introduction.md

**Comment by rijkvanzanten** [comment]
Heya! You're quoting the docs for the previous version :)

V9s graphql endpoint is fully functional and available. The v9 docs lists al functionality in both rest and graphql:
https://docs.directus.io/reference/api/items/

**Reply by rijkvanzanten** [reply]
Oh didn't see @benhaynes's answer on mobile ^^

**Reply by martin-braun** [reply]
Thank you very much! Yes I am, that happens when you use DuckDuckGo, instead of Google. Maybe it makes sense to add a link to the v9 docs in the header of the v8 docs, to prevent confusion?

**Comment by rijkvanzanten** [comment]
~~Heya! Thank you for taking the time to submit this request!~~

~~It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:~~

~~This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!~~

~~_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._~~

**Reply by martin-braun** [reply]
Your bot engine is out of control, this is already closed. 😀

**Reply by rijkvanzanten** [reply]
@martin-braun I'm gonna tell you something you won't believe: I manually copy pasted that message across thousands. GitHub's API doesn't support updating/closing PRs yet 😵 

The out of control part of it is literally just my brain not keeping up hahah

**Comment by rijkvanzanten** [comment]
Permissions and mutations exist! Subscriptions are coming very soon 🙂
